frontend/src/components/InstanceForm/Buttons.vue

```diff
@@ -292,7 +292,21 @@ const doCreate = async () => {
 const updateEditState = (instance: Instance) => {
   basicInfo.value = extractBasicInfo(instance);
   const updatedEditState = extractDataSourceEditState(instance);
+  
+  // Explicitly preserve extraConnectionParameters from the instance when setting the updated edit state
+  if (instance.dataSources) {
+    instance.dataSources.forEach((originalDs) => {
+      const ds = updatedEditState.dataSources.find(d => d.id === originalDs.id);
+      if (ds && originalDs.extraConnectionParameters) {
+        // Ensure we have a fresh copy of the extraConnectionParameters
+        ds.extraConnectionParameters = { ...originalDs.extraConnectionParameters };
+      }
+    });
+  }
```

Comment:
This block (line 296-line 305) should put into the `extractDataSourceEditState`. Same for https://github.com/bytebase/bytebase/pull/15419/files#diff-05d2048496738439aae78c017674d35b106a2dd030f8640b5f972d015784313cR112